### ISSUE TITLE
[Refactor][Op] Split decode kernels into no-split, splitKV, and combine

### ISSUE THREAD

**Issue by superAngGao** [tile-ai/TileOPs#722]
## Summary

Refactor existing GQA/MHA decode kernels to separate the no-split and split-KV code paths into independent kernel classes, plus a shared combine kernel.

Parent issue: #719

## Motivation

Current decode kernels auto-dispatch split vs no-split internally within a single class. This causes:
- Mixed grid dimensions and output layouts in one kernel
- Different autotune search spaces conflated
- Code duplication when adding feature combinations (each feature must implement both paths)
- Potential bugs during kernel registration

## Current state

| Kernel class | Internal paths |
|-------------|---------------|
| `gqa_decode_kernel` | no-split + split+combine |
| `gqa_decode_paged_kernel` | no-split + split+combine |
| `mha_decode_kernel` | no-split + split+combine |
| `mha_decode_paged_kernel` | no-split + split+combine |

The split kernel and combine are already separate macros (`_gqa_decode_split` + `combine`) inside the same `prim_func`. This refactor promotes them to independent kernel classes.

## Target state

| New kernel class | Responsibility |
|-----------------|---------------|
| `GQADecodeKernel` | No-split, short sequences |
| `GQADecodeSplitKVKernel` | Split, outputs partial_o + partial_lse |
| `GQADecodeCombineKernel` | Merges partial results via online softmax (shared) |
| `GQADecodePagedKernel` | No-split paged |
| `GQADecodePagedSplitKVKernel` | Split paged |
| `GQADecodePagedCombineKernel` | Combine paged (shared) |
| `MHADecodeKernel` | No-split |
| `MHADecodeSplitKVKernel` | Split |
| `MHADecodeCombineKernel` | Combine (shared) |
| `MHADecodePagedKernel` | No-split paged |
| `MHADecodePagedSplitKVKernel` | Split paged |
| `MHADecodePagedCombineKernel` | Combine paged (shared) |

Also renames snake_case kernel classes to PascalCase per naming convention.

### `num_split` alignment between split and combine

`num_split` is determined by the split kernel's autotune config. The Op reads `num_split` from the split kernel's config after autotune and uses it to construct the combine kernel:

```python
class GQADecodeOp(Op):
    def __init__(self, batch, heads, heads_kv, seqlen_kv, dim, dtype, tune=False):
        self.no_split_kernel = GQADecodeKernel(batch, heads, heads_kv, seqlen_kv, dim, dtype)
        self.split_kernel = GQADecodeSplitKVKernel(batch, heads, heads_kv, seqlen_kv, dim, dtype, tune=tune)
        
        # After autotune, read num_split from split kernel's config
        num_split = self.split_kernel.config["num_split"]
        self.combine_kernel = GQADecodeCombineKernel(batch, heads, dim, num_split, dtype)
        
        # Pre-allocate intermediate tensors
        self.glse = torch.empty(batch, heads, num_split, dtype=dtype, device="cuda")
        self.partial_o = torch.empty(batch, heads, num_split, dim, dtype=dtype, device="cuda")
```

### Autotune strategy

- **Split kernel**: autotunes over `{block_H, block_N, num_split, num_stages, threads}` (same as current)
- **Combine kernel**: no autotune needed — it is a simple reduction over `num_split` partial results, performance is not sensitive to tuning
- **No-split kernel**: autotunes over `{block_H, block_N, num_stages, threads}` independently

### Intermediate tensor pre-allocation

`glse` and `Output_partial` shapes depend on `num_split`. Currently they are allocated in `forward()` every call. After refactor, the Op pre-allocates them in `__init__()` based on the autotuned `num_split`:

```python
# Pre-allocated in __init__:
self.glse = torch.empty(batch, heads, num_split, dtype=dtype, device="cuda")
self.partial_o = torch.empty(batch, heads, num_split, dim, dtype=dtype, device="cuda")
```

### Combine kernel is universal

The combine kernel only depends on `{batch, heads, num_split, dim}` and performs online softmax merging. It does **not** depend on any attention feature (RoPE, FP8, LogitsClip, etc.), so a single combine kernel class is shared across all feature combinations.

### Op dispatch (runtime)

```python
class GQADecodeOp(Op):
    def forward(self, Q, K, V, real_seqlen_kv):
        num_split = self.split_kernel.config["num_split"]
        block_N = self.split_kernel.config["block_N"]

        if real_seqlen_kv < num_split * block_N:
            return self.no_split_kernel(Q, K, V, real_seqlen_kv)

        self.split_kernel(Q, K, V, real_seqlen_kv, self.glse, self.partial_o)
        return self.combine_kernel(self.partial_o, self.glse)
```

### Split kernel output contract

| Output | Shape | Description |
|--------|-------|-------------|
| `partial_o` | `[batch, heads, num_split, dim]` | Per-split weighted output (already divided by local logsum) |
| `glse` | `[batch, heads, num_split]` | Per-split log-sum-exp in log2 space: `log2(logsum) + scores_max * scale` |

The combine kernel uses these to reconstruct the final output via:
```
lse_max = max(glse[..., :])
w_k = exp2(glse[..., k] - lse_logsum)
output = sum(w_k * partial_o[..., k, :])
```

## Scope

- 4 existing kernel classes → 12 kernel classes
- No new features, pure refactor
- Tests and benchmarks must continue to pass
- This is a prerequisite for #719 feature kernels (RoPE, FP8, etc.)
